crates/wysiwyg/src/composer_model/mentions.rs

```diff
@@ -13,15 +13,43 @@
 // limitations under the License.
 
 use crate::{
-    dom::{nodes::MentionNode, DomLocation},
-    ComposerModel, ComposerUpdate, DomNode, Location, SuggestionPattern,
-    UnicodeString,
+    dom::{
+        nodes::{MentionNode, MentionNodeKind},
+        DomLocation,
+    },
+    ComposerModel, ComposerUpdate, DomNode, Location, MentionsState,
+    SuggestionPattern, UnicodeString,
 };
 
 impl<S> ComposerModel<S>
 where
     S: UnicodeString,
 {
+    /// Returns the current mentions state of the content of the RTE editor.
+    pub fn get_mentions_state(&self) -> MentionsState {
+        let mut mentions_state = MentionsState::default();
+        for node in self.state.dom.iter_mentions() {
+            match node.kind() {
+                MentionNodeKind::AtRoom => {
+                    mentions_state.has_at_room_mention = true
+                }
+                MentionNodeKind::MatrixUri { mention } => {
+                    match mention.kind() {
+                        matrix_mentions::MentionKind::Room => {
+                            // Not required yet
+                        }
```

Comment:
The omission of rooms in the result could be a bit confusing given the name of the function because it sounds like it will return all mentions. Can I suggest either:
- Update the name of the function and result to `get_intentional_mentions() -> IntentionalMentions` and add documentation that references [MSC3952](https://github.com/matrix-org/matrix-spec-proposals/pull/3952) so that it's implied that only spec'ed mentions will be returned. Or,
- Include room IDs in the result and leave it up to the client to ignore them.

Reply:
Adding a way to also include `room_ids` and `room_aliases`